src/context.rs

```diff
@@ -44,4 +52,27 @@ impl Context {
 
         Ok(::device::new(self, ptr))
     }
+
+    /// Creates a device for a given major/minor device number pair
+    pub fn device_from_devnum(&self, kind: ::DeviceType, major: u64, minor: u64) -> ::Result<Device> {
+        let devnum = makedev(major, minor);
+        let ptr = try_alloc!(unsafe {
+            ::ffi::udev_device_new_from_devnum(self.udev, match kind {
+                ::DeviceType::Char => b'c' as ::libc::c_char,
```

Comment:
I think it might help to abstract this translation from DeviceType to string code into a function and make it part of DeviceType impl.